### ISSUE TITLE
Icon aria-label should be configurable

### ISSUE THREAD

**Issue by danethurber** [pluralsight/design-system#462]
from @jim-cooper 

| We noticed that the `Icon` doesn't have a title property.  On the course table of contents where we show checkmark icons; for screen readers we'd love to have it say "Module Complete" instead of "Check Icon Image".  Is there a good way to do this currently?

### TODO

add prop to make this configurable
